webapp/src/components/App/Singlestudy/explore/Modelization/Map/index.tsx

```diff
@@ -300,19 +301,32 @@ function Map() {
               };
             });
             setNodeData(tempNodeData);
+            const links = await getAllLinks({ uuid: study.id, withUi: true });
             setLinkData(
-              Object.keys(data.areas).reduce(
-                (links, currentAreaId) =>
-                  links.concat(
-                    Object.keys(data.areas[currentAreaId].links).map(
-                      (linkId) => ({
-                        source: currentAreaId,
-                        target: linkId,
-                      })
-                    )
-                  ),
-                [] as Array<LinkProperties>
-              )
+              links.map((link) => {
+                let style = [0];
+                let linecap = "butt";
+                if (link.ui?.style === "dot") {
+                  style = [1, 5];
+                  linecap = "round";
+                }
+                if (link.ui?.style === "dash") {
+                  style = [16, 8];
+                  linecap = "square";
+                }
+                if (link.ui?.style === "dotdash") {
+                  style = [10, 6, 1, 6];
+                  linecap = "square";
+                }
```

Comment:
here you can use R.cond from Ramda (check examples in code) : https://ramdajs.com/docs/#cond